File: client/java/src/avt/Scroll.java
package avt;

import javax.microedition.lcdui.Graphics;
import main.Canvas;

public final class Scroll {
   private static Scroll instance;
   private int limit;
   private int temp;
   private int dis;
   private int yScroll;
   private int hScroll;
   private int h;
   private boolean Disvisible = false;

   public static Scroll gI() {
      return instance == null ? (instance = new Scroll()) : instance;
   }

   public final void init(int var1, int var2) {
      if (var1 < var2) {
         this.Disvisible = true;
      } else {
         this.Disvisible = false;
         this.temp = var1;
         this.limit = var2;
         this.dis = var2 * var2 / var1;
         if (this.dis <= 0) {
            this.dis = 1;
         }
      }

   }

   public final void updateScroll(int var1, int var2) {
      this.hScroll = var1;
      this.h = var2;
      if (!this.Disvisible && (CRes.abs(var2 - var1) > 5 || Canvas.cameraList.n != 0 || Canvas.cameraList.p)) {
         var2 = this.temp * 100 / 100;
         var1 = var1 * 100 / var2;
         var2 = this.limit * 100 / 100;
         this.yScroll = var1 * var2;
      }

   }

   public final void paintScroll(Graphics var1, int var2, int var3) {
      if (!this.Disvisible && (CRes.abs(this.h - this.hScroll) > 5 || Canvas.cameraList.n != 0 || Canvas.cameraList.p)) {
         var1.setColor(6201499);
         var1.setClip(var2 - 1, -1, 6, this.limit + 2);
         var1.fillRect(var2, 0 + this.yScroll / 100, 4, this.dis);
      }

   }
}

File: client/unity/Assets/Scripts/Assembly-CSharp/Scroll.cs
using System;

public class Scroll
{
	private static Scroll instance;

	public int limit;

	public int temp;

	public int dis;

	public int yScroll;

	public int hScroll = 100;

	public int h;

	public static bool Disvisible;

	public static bool isAble;

	private FrameImage img;

	public Scroll()
	{
		if (AvMain.hd == 2)
		{
			img = new FrameImage(Image.createImagePNG(T.getPath() + "/temp/scrollList"), 7, 20);
		}
		else
		{
			img = new FrameImage(Image.createImagePNG(T.getPath() + "/temp/scrollList"), 4, 10);
		}
	}

	public static Scroll gI()
	{
		if (instance == null)
		{
			instance = new Scroll();
		}
		return instance;
	}

	public void init(int dis, int size, int cmy)
	{
		if (size == 0)
		{
			size = 1;
		}
		if (limit == 0)
		{
			limit = 1;
		}
		try
		{
			this.dis = dis;
			hScroll = 100;
			limit = size - dis;
			if (size > dis)
			{
				hScroll = dis * 100 / size;
				if (hScroll < 2)
				{
					hScroll = 2;
				}
				h = cmy * 100 / limit;
				int num = dis - dis * hScroll / 100;
				yScroll = h * num / 100;
			}
			Disvisible = limit + dis < dis;
			isAble = true;
		}
		catch (Exception)
		{
		}
	}

	public void updateScroll(int cmy, int cmtoY, int v)
	{
		if (!Disvisible && limit != 0)
		{
			if (cmtoY < 0)
			{
				cmtoY = 0;
			}
			h = cmy * 100 / limit;
			int num = dis * hScroll / 100;
			if (num < 22 * AvMain.hd)
			{
				num = 22 * AvMain.hd;
			}
			int num2 = dis - num;
			yScroll = h * num2 / 100;
			if (yScroll > dis - 3)
			{
				yScroll = dis - 3;
			}
			if (yScroll < 0)
			{
				yScroll = 0;
			}
			if (yScroll + num > dis)
			{
				yScroll = dis - num;
			}
			if (cmy != cmtoY || Canvas.isPointerDown || v != 0)
			{
				isAble = true;
			}
			else
			{
				isAble = false;
			}
		}
	}

	public void paintScroll(MyGraphics g, int x, int y)
	{
		if (!Disvisible && isAble)
		{
			int num = dis * hScroll / 100;
			if (num < 22 * AvMain.hd)
			{
				num = 22 * AvMain.hd;
			}
			g.setClip(x, y + yScroll, 7f, num);
			img.drawFrame(0, x, y + 1 + yScroll, 0, g);
			img.drawFrame(2, x, y + 1 + yScroll + num - 2 - 10 * AvMain.hd, 0, g);
			int num2 = (num - 20 * AvMain.hd) / (10 * AvMain.hd);
			for (int i = 0; i < num2; i++)
			{
				img.drawFrame(1, x, y + 1 + yScroll + 10 * AvMain.hd + 10 * AvMain.hd * i, 0, g);
			}
			int num3 = num - 2 - 10 * AvMain.hd - (10 * AvMain.hd + num2 * 10 * AvMain.hd);
			g.drawRegion(img.imgFrame, 0f, 10 * AvMain.hd, 7, num3, 0, x, y + 1 + yScroll + 10 * AvMain.hd + num2 * 10 * AvMain.hd, 0);
		}
	}
}
